workflows/epigenetics/chipseq-pe-with-replicates-controls/test-data/test_sample_sheet.tsv

```diff
@@ -0,0 +1,4 @@
+SRR5204807	Spt5	rep1	SRR5204809
```

Comment:
Can we not create this file dynamically by having the users provide sample and ip collections ? I don't hate this for running this on the command line, but a lot can and will go wrong here. You mentioned issues if control and ip numbers differ, but I think that would also still be the case with the sample sheet.

Reply:
One of the issues is that with the sample sheet you can define the same sample to be the appropriate control for several samples in a replicate, or each sample can have its own control, or whatever. In the sample sheet this connection is made via columns 1 and 4, but you cannot do this flexibly with just collections (unless you rely on naming schemes of elements, which is even worse than the sample sheet approach).
Averaging across replicates is also harder and more error-prone without a sample sheet, which establishes nicely which samples represent the same condition across replicates (maybe look at the example from the README).

Reply:
Can you make it two input collections, and if you need to repeat a control .... you repeat the control ? I don't see the difficulty currently. I would currently not let users build nested collections.

Reply:
Replicates are determined by the order of the collection elements, not the identifier

Reply:
Note that in the above sample A and B, for example, are not IPed sample and control, but two different IPs done on the same input.

Reply:
Can we say one contrast per analysis for a start ? If you want to use a sample sheet you can have a parent workflow align your inputs according to the sheet ? I think that would nicely encapsulate the complexity while maintaining a easy to use workflow for the common case of doing a single condition ?

Reply:
Hmm, that could be a way forward, but not for tonight? Quite some rewiring, tests rewriting, and manual testing, plus readme rewrite that's required for this.
As I said before, both @lldelisle 's chipseq_pe and/or cutandrun WFs are a good starting point for tomorrow, too.